### PR TITLE
Hdrp/refactor hdr passet inspector

### PR DESCRIPTION
**Pull request by RSlysz**
### Purpose of this PR
Cleaned RenderPipelineSettings usage confusion and ensure runtime immutability
Update HDRPAsset inspector (And merged numerous files used to draw it)
Add multi edition support for HDRPAsset
Fix newly spotted multi edition issue on FrameSettings

---
### Release Notes
Added multi edition support for HDRenderPipelineAsset.
Fixed multi edition issue in FrameSettings.
Updated HDRenderPipelineAsset inspector.

---
### Testing status
**Katana Tests**: Running: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP%2Frefactor-HDRPasset-inspector&unity_branch=trunk&automation-tools_branch=add-platform-filter

**Manual Tests**: Tested locally, Win Player build successfully

**Automated Tests**: NA

---
### Overall Product Risks
**Technical Risk**: High (Many files impacted)

**Halo Effect**: Medium-High (RenderPipelineSettings are base of many things)

---
### Comments to reviewers
@alelievr : Added mainly for the merge shadow in lighting as it is a task you'll be on.
@fredpointzero : Added mainly for data structure change for RenderPipelineSettings.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/RenderPipelineSettings.Migration.cs.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/RenderPipelineSettings.Migration.cs.cs]: the file have .cs.cs extension, not expected

**Comment by sebastienlagarde** [comment]
I merged #2739, you will be require to merge and resolve conflict (maybe the simpler is to see the tooltips correction done and just report them on the modified file, then discard the conflict

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/OverridableFrameSettingsArea.cs [hunk 2]
```diff
@@ -190,10 +190,11 @@ void DrawField(Field field, bool withOverride)
                                     break;
                                 case FrameSettingsFieldAttribute.DisplayType.BoolAsEnumPopup:
                                     //shame but it is not possible to use Convert.ChangeType to convert int into enum in current C#
+                                    //Also, Enum.Equals and Enum operator!= always send true here. Rely on GetHashCode to obtain index in enum.
                                     //rely on string parsing for the moment
                                     var oldEnumValue = Enum.Parse(attributes[field.field].targetType, serializedFrameSettings.IsEnabled(field.field) ? "1" : "0"); 
-                                    var newEnumValue = (Enum)DrawFieldShape(field.label, oldEnumValue);
-                                    if (oldEnumValue != newEnumValue)
+                                    var newEnumValue = DrawFieldShape(field.label, oldEnumValue).GetHashCode();
```
Comment (fredpointzero) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/OverridableFrameSettingsArea.cs]: Try to use this code:

```
var oldEnumValue = Enum.GetValues(targetType).GetValue(serializedFrameSettings.IsEnabled(field.field)? 1 : 0);
var newEnumValue = Convert.ChangeType(DrawFieldShape(field.label, oldEnumValue), attributes[field.field].targetType);
var oldEnumIntValue = ((IConvertible)oldEnumValue).ToInt32(System.Globalization.CultureInfo.CurrentCulture);
var newEnumIntValue = ((IConvertible)newEnumValue ).ToInt32(System.Globalization.CultureInfo.CurrentCulture);
```

File: com.unity.render-pipelines.high-definition/CHANGELOG.md [hunk 3]
```diff
@@ -46,6 +46,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Added support for "After Post-Process" render pass for unlit shader
 - Added support for textured rectangular area lights
 - Added stereo instancing macros to MSAA shaders
+- Added multi edition support for HDRenderPipelineAsset
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/CHANGELOG.md]: please move the new line at the end of the list. Thanks, same for the other

**Comment by RSlysz** [comment]
@alelievr reverted merge of shadow structure in lighting due to heavy weight on assets and unability to cleanly migrate inside struct. You will need to keep both.

@sebastienlagarde Locally, only 2701 (sometimes) and 1402 (always) fails as in master. Katana running again to be sure.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/OverridableFrameSettingsArea.cs (hunk 2 above)
Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/OverridableFrameSettingsArea.cs]: @alelievr FYI: This is way cleaner as GetHashCode could be platform dependent.

**Pull request merged**